WordPress/gutenberg
# Apply ! important to user preset styles for theme.json themes only

This is a proposed interim solution to the [issues outlined here](https://github.com/WordPress/gutenberg/issues/37590), with the view to exploring removal of the `! important` from user presets across all themes.

## What?
Limits the addition of `! important` to user preset property values to themes with a `theme.json` file. The problems outlined by theme authors were all related to Classic themes (that we are aware of), so this change puts Classic themes back to the pre 5.9 state. A Classic theme can still opt in to have the `! important` applied to user presets by adding a theme.json file to the theme.

## Why?
The blanket addition of  `! important` to these property values [causes some issues for theme author](https://github.com/WordPress/gutenberg/issues/37590)s, particularly with the likes of responsive font sizing.

## How?
Checks for the existence of a `theme.json` file before applying `! important`

## Testing Instructions

- With WordPress 5.9.2 install [Michelle 1.2.0](https://downloads.wordpress.org/theme/michelle.1.2.0.zip).
- Create a page and add a [content from this gist](https://gist.github.com/webmandesign/8136127b6f3ec52d53b253689f95a248).
- Publish the page and check how it looks on frontend. The font size in both columns should be equal but if you reduce the screen size the font sizes between columns become uneven because in the left column the `! important` in the GB presets is overriding the responsive typography  CSS
- Now run the same with this branch checked out and built and the fonts should resize in both columns as expected.
- Select TT2 theme and add a paragraph and set the typography size. Check frontend and make sure the font size class has `! important` set, eg.
```css
has-large-font-size {
    font-size: var(--wp--preset--font-size--large)!important;
}
```
